File: PB - More Exercises Python/5.0 For-Loop - More Exercises/03. Logistics.py
cargo_count = int(input())
cargos_dict = {'all_cargo' : 0, 'bus': 0, 'truck' : 0, 'train' : 0}

for _ in range(cargo_count):
    current_cargo = int(input())
    if current_cargo <= 3:
        cargos_dict['bus'] += current_cargo
    elif 4 <= current_cargo <= 11:
        cargos_dict['truck'] += current_cargo
    else:
        cargos_dict['train'] += current_cargo
    cargos_dict['all_cargo'] += current_cargo

average_price = (cargos_dict['bus'] * 200 + cargos_dict['truck'] * 175 + cargos_dict['train'] * 120) / cargos_dict['all_cargo']

print(f"{average_price:.2f}")
print(f"{cargos_dict['bus'] / cargos_dict['all_cargo'] *  100:.2f}%")
print(f"{cargos_dict['truck'] / cargos_dict['all_cargo'] * 100:.2f}%")
print(f"{cargos_dict['train'] / cargos_dict['all_cargo'] * 100:.2f}%")


'''
4
1
5
16
3
'''

File: PB - More Exercises GO/5.0 For-Loop - More Exercises/03. Logistics.go
package main

import "fmt"

func main() {
	var cargoCount int
	fmt.Scan(&cargoCount)
	cargosObj := map[string]float64{"allCargo": 0.0, "bus": 0.0, "truck": 0.0, "train": 0.0}

	for i := 0; i < cargoCount; i++ {
		var currentCargo float64
		fmt.Scan(&currentCargo)

		if currentCargo <= 3 {
			cargosObj["bus"] += currentCargo
		} else if 4 <= currentCargo && currentCargo <= 11 {
			cargosObj["truck"] += currentCargo
		} else {
			cargosObj["train"] += currentCargo
		}

		cargosObj["allCargo"] += currentCargo
	}

	averagePrice := (cargosObj["bus"]*200 + cargosObj["truck"]*175 + cargosObj["train"]*120) / cargosObj["allCargo"]

	fmt.Printf("%.2f\n", float64(averagePrice))
	fmt.Printf("%.2f%%\n", cargosObj["bus"]/cargosObj["allCargo"]*100)
	fmt.Printf("%.2f%%\n", cargosObj["truck"]/cargosObj["allCargo"]*100)
	fmt.Printf("%.2f%%\n", cargosObj["train"]/cargosObj["allCargo"]*100)
}

/*
name   :03. Logistics
input  :4 1 5 16 3
output :
143.80
16.00%
20.00%
64.00%
*/
